server/src/main/java/io/druid/segment/realtime/appenderator/StreamAppenderatorDriver.java

```diff
@@ -378,10 +396,23 @@ public void onFailure(Throwable e)
       final Collection<String> sequenceNames
   )
   {
-    return Futures.transform(
-        publish(publisher, committer, sequenceNames),
-        this::registerHandoff
-    );
+    final SettableFuture<SegmentsAndMetadata> publishDoneFuture = SettableFuture.create();
+    final ListenableFuture<SegmentsAndMetadata> publishFuture = publish(publisher, committer, sequenceNames);
```

Comment:
Unless there is a bug, as it is currently written, there is no point in creation of `publishDoneFuture` and `publishFuture` could be just returned from this method.

Reply:
bug, fixing